src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -200,14 +200,16 @@ public void Enqueue(TElement element, TPriority priority)
             // Note that the node being enqueued does not need to be physically placed
             // there at this point, as such an assignment would be redundant.
 
-            int currentSize = _size++;
+            int currentSize = _size;
             _version++;
 
             if (_nodes.Length == currentSize)
             {
                 Grow(currentSize + 1);
             }
 
+            _size = currentSize + 1;
```

Comment:
previously you passed in currentSize to MoveUpDefaultComparer/MoveUpCustomComparer and now this value is smaller by 1. I'm not seeing any logic adjustment - is that by design?